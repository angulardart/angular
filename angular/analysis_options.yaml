# This is a copy of ../analysis_options.yaml, with implicit-casts: false, and an "exclude: ...".

analyzer:
  strong-mode:
    implicit-casts: false
  exclude:
    - "tools/analyzer_plugin/bin/plugin.dart"
  errors:
    unused_element: error
    unused_import: error
    unused_local_variable: error
    dead_code: error

    # Allow importing .template.dart files without an analyzer error.
    uri_has_not_been_generated: ignore

    # The noise of deprecations isn't worth the effort.
    deprecated_member_use_from_same_package: ignore
linter:
  rules:
    - avoid_empty_else
    - avoid_init_to_null
    - avoid_null_checks_in_equality_operators
    - await_only_futures
    - camel_case_types
    - cancel_subscriptions
    - control_flow_in_finally
    - directives_ordering
    - empty_catches
    - empty_constructor_bodies
    - empty_statements
    - hash_and_equals
    - iterable_contains_unrelated_type
    - library_names
    - library_prefixes
    - list_remove_unrelated_type
    - only_throw_errors
    - overridden_fields
    - package_api_docs
    - package_names
    - package_prefixed_library_names
    - prefer_final_fields
    - prefer_is_not_empty
    - slash_for_doc_comments
    - test_types_in_equals
    - throw_in_finally
    - type_init_formals
    - unnecessary_brace_in_string_interps
    - unrelated_type_equality_checks
    - valid_regexps
