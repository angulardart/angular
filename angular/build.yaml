targets:
  $default:
    builders:
      angular:
        generate_for:
          exclude:
            - "lib/src/build.dart"
            - "lib/src/compiler/**"
            - "lib/src/source_gen/**"

builders:
  angular:
    import: "package:angular/src/build.dart"
    builder_factories:
      - templatePlaceholder
      - templateCompiler
      - stylesheetCompiler
    auto_apply: dependents
    applies_builders:
      - "angular|placeholder_cleanup"
      - "angular|component_source_cleanup"
    # See https://github.com/dart-lang/angular/issues/988.
    is_optional: true
    required_inputs:
      - ".css"
    build_extensions:
      .css:
        - ".css.dart"
        - ".css.shim.dart"
      .dart:
        - ".template.dart"
post_process_builders:
  placeholder_cleanup:
    import: "package:angular/src/build.dart"
    builder_factory: "placeholderCleanup"
  component_source_cleanup:
    import: "package:angular/src/build.dart"
    builder_factory: "componentSourceCleanup"
    defaults:
      generate_for: ["lib/**"]
      release_options:
        enabled: true
