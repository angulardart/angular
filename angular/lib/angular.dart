/// The primary library for the [AngularDart web framework][AngularDart].
///
/// Import this library as follows:
///
/// ```
/// import 'package:angular/angular.dart';
/// ```
///
/// For help using this library, see the AngularDart documentation:
///
/// * [AngularDart guide][]
/// * [AngularDart cheat sheet][cheatsheet]
///
/// [AngularDart]: https://webdev.dartlang.org/angular
/// [AngularDart guide]: https://webdev.dartlang.org/angular/guide
/// [cheatsheet]: https://webdev.dartlang.org/angular/cheatsheet

library angular;

export 'core.dart';
export 'src/common/common_directives.dart';
export 'src/common/directives.dart';
export 'src/common/pipes.dart';
export 'src/core/angular_entrypoint.dart' show AngularEntrypoint;
export 'src/core/application_tokens.dart' hide APP_ID_RANDOM_PROVIDER;
export 'src/platform/bootstrap.dart';
export 'src/platform/dom/dom_tokens.dart';
export 'src/platform/dom/events/event_manager.dart' show EventManagerPlugin;
