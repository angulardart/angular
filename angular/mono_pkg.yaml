dart:
- stable
- dev

stages:
- analyze:
  - dartanalyzer: --fatal-warnings .
