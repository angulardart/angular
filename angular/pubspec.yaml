name: angular
version: 5.3.0
author: Dart Team <web@dartlang.org>
description: Fast and productive web framework
homepage: https://webdev.dartlang.org/angular
documentation: https://webdev.dartlang.org/angular/api
environment:
  sdk: ">=2.2.1-dev.3.0 <3.0.0"

dependencies:
  # Runtime (or runtime *and* the compiler).
  collection: ^1.14.5
  intl: '>=0.13.0 <0.16.0'
  js: ^0.6.0
  logging: '>=0.9.0 <0.12.0'
  meta: ^1.1.4
  stack_trace: ^1.9.1

  # Compiler. Eventually we want to move these to angular_compiler.
  analyzer: ^0.36.2

  # It is extremely important these two dependencies stay pinned to current.
  angular_ast: 0.5.9
  angular_compiler: 0.4.3
  ##########################################################################

  build: '>=0.12.7 <2.0.0'
  build_config: '>=0.2.6 <0.5.0' # Runtime dependency with no import
  code_builder: '^3.0.1'
  csslib: '>=0.14.5 <0.16.0'
  path: ^1.6.1
  source_gen: ">=0.9.1 <0.10.0"
  source_span: ^1.4.0

# === vvv REMOVE WHEN PUBLISHING vvv ===
dependency_overrides:
  angular_ast:
      path: ../angular_ast
  angular_compiler:
      path: ../angular_compiler
# === ^^^ REMOVE WHEN PUBLISHING ^^^ ===
