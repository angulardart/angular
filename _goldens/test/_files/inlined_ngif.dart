import 'package:angular/angular.dart';

/// Tests that this simple NgIf is inlined.
@Component(
  selector: 'inlined-ngif',
  template: '''
    <div *ngIf="maybeTrue">Hello World!</div>
    ''',
  directives: [NgIf],
)
class InlinedNgIfComponent {
  var maybeTrue = true;
}

/// Tests that this simple NgIf is *not* inlined.
@Component(
  selector: 'inlined-ngif',
  template: '''
    <div *ngIf="name == 'World'">Hello {{name}}!</div>
    ''',
  directives: [NgIf],
)
class IgnoredNgIfComponent {
  final name = 'World';
}

/// Tests that an *ngIf that is always true is just treated as plain HTML.
@Component(
  selector: 'inlined-ngif-with-immutable-condition',
  template: '''
    <div *ngIf="alwaysTrue">Hello World!</div>
    ''',
  directives: [NgIf],
)
class InlinedNgIfWithImmutableConditionComponent {
  final alwaysTrue = true;
}
