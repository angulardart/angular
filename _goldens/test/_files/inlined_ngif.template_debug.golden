// **************************************************************************
// Generator: Instance of 'Compiler'
// **************************************************************************

// ignore_for_file: cancel_subscriptions,constant_identifier_names,duplicate_import,non_constant_identifier_names,library_prefixes,UNUSED_IMPORT,UNUSED_SHOWN_NAME
import 'inlined_ngif.dart';
export 'inlined_ngif.dart';
import 'package:angular/angular.dart';
import 'package:angular/src/di/reflector.dart' as _ngRef;
import 'package:angular/angular.template.dart' as _ref0;
import 'package:angular/src/debug/debug_context.dart';
import 'package:angular/src/common/directives/ng_if.dart';
import 'package:angular/src/debug/debug_app_view.dart';
import 'inlined_ngif.dart' as import3;
import 'dart:html' as import4;
import 'package:angular/src/core/render/api.dart';
import 'package:angular/src/core/linker/app_view.dart';
import 'package:angular/src/core/linker/view_type.dart' as import7;
import 'package:angular/src/core/change_detection/change_detection.dart';
import 'package:angular/src/core/linker/app_view_utils.dart' as import9;
import 'package:angular/angular.dart';

const List<dynamic> styles$InlinedNgIfComponent = const [];
List<StaticNodeDebugInfo> nodeDebugInfos_InlinedNgIfComponent0 = [
  new StaticNodeDebugInfo([NgIf], null, <String, dynamic>{})
];

class ViewInlinedNgIfComponent0 extends DebugAppView<import3.InlinedNgIfComponent> {
  import4.Comment _anchor_0;
  import4.DivElement _el_0_0;
  import4.Text _text_0_1;
  bool _expr_0 = false;
  static RenderComponentType _renderType;
  ViewInlinedNgIfComponent0(AppView<dynamic> parentView, int parentIndex) : super(import7.ViewType.COMPONENT, {}, parentView, parentIndex, ChangeDetectionStrategy.CheckAlways, nodeDebugInfos_InlinedNgIfComponent0) {
    rootEl = import4.document.createElement('inlined-ngif');
    _renderType ??= import9.appViewUtils.createRenderType('asset:_goldens/test/_files/inlined_ngif.dart class InlinedNgIfComponent - inline template', ViewEncapsulation.None, styles$InlinedNgIfComponent);
    setupComponentType(_renderType);
  }
  @override
  ComponentRef<import3.InlinedNgIfComponent> build() {
    final _rootEl = rootEl;
    final import4.HtmlElement parentRenderNode = initViewRoot(_rootEl);
    _anchor_0 = ngAnchor.clone(false);
    parentRenderNode.append(_anchor_0);
    dbgElm(this, _anchor_0, 0, 0, 5);
    init([], null, [_anchor_0]);
    return null;
  }

  @override
  void detectChangesInternal() {
    final import3.InlinedNgIfComponent _ctx = ctx;
    dbg(0, 0, 5);
    final currVal_0 = (_ctx.foo == true);
    if (import9.checkBinding(_expr_0, currVal_0)) {
      if (currVal_0) {
        var doc = import4.document;
        _el_0_0 = doc.createElement('div');
        dbgElm(this, _el_0_0, 0, 0, 0);
        _text_0_1 = new import4.Text('Hello World!');
        _el_0_0.append(_text_0_1);
        dbgElm(this, _text_0_1, 1, 0, 17);
        addInlinedNodes(_anchor_0, [_el_0_0], true);
      } else {
        removeInlinedNodes([_el_0_0], true);
      }
      _expr_0 = currVal_0;
    }
  }
}

AppView<import3.InlinedNgIfComponent> viewFactory_InlinedNgIfComponent0(AppView<dynamic> parentView, int parentIndex) {
  return new ViewInlinedNgIfComponent0(parentView, parentIndex);
}

const List<dynamic> styles$InlinedNgIfComponentHost = const [];
List<StaticNodeDebugInfo> nodeDebugInfos_InlinedNgIfComponentHost0 = [
  new StaticNodeDebugInfo([import3.InlinedNgIfComponent], import3.InlinedNgIfComponent, <String, dynamic>{})
];

class _ViewInlinedNgIfComponentHost0 extends DebugAppView<dynamic> {
  ViewInlinedNgIfComponent0 _compView_0;
  import3.InlinedNgIfComponent _InlinedNgIfComponent_0_5;
  _ViewInlinedNgIfComponentHost0(AppView<dynamic> parentView, int parentIndex) : super(import7.ViewType.HOST, {}, parentView, parentIndex, ChangeDetectionStrategy.CheckAlways, nodeDebugInfos_InlinedNgIfComponentHost0);
  @override
  ComponentRef build() {
    _compView_0 = new ViewInlinedNgIfComponent0(this, 0);
    rootEl = _compView_0.rootEl;
    dbgIdx(rootEl, 0);
    _InlinedNgIfComponent_0_5 = new import3.InlinedNgIfComponent();
    _compView_0.create(_InlinedNgIfComponent_0_5, projectableNodes);
    init0Dbg(rootEl, [rootEl]);
    return new ComponentRef<import3.InlinedNgIfComponent>(0, this, rootEl, _InlinedNgIfComponent_0_5);
  }

  @override
  void detectChangesInternal() {
    _compView_0.detectChanges();
  }

  @override
  void destroyInternal() {
    _compView_0?.destroy();
  }
}

AppView viewFactory_InlinedNgIfComponentHost0(AppView<dynamic> parentView, int parentIndex) {
  return new _ViewInlinedNgIfComponentHost0(parentView, parentIndex);
}

const ComponentFactory<import3.InlinedNgIfComponent> InlinedNgIfComponentNgFactory = const ComponentFactory<import3.InlinedNgIfComponent>('inlined-ngif', viewFactory_InlinedNgIfComponentHost0, _InlinedNgIfComponentMetadata);
const List<dynamic> styles$InlinedNgIfWithImmutableConditionComponent = const [];
List<StaticNodeDebugInfo> nodeDebugInfos_InlinedNgIfWithImmutableConditionComponent0 = [
  new StaticNodeDebugInfo([NgIf], null, <String, dynamic>{})
];

class ViewInlinedNgIfWithImmutableConditionComponent0 extends DebugAppView<import3.InlinedNgIfWithImmutableConditionComponent> {
  import4.Comment _anchor_0;
  import4.DivElement _el_0_0;
  import4.Text _text_0_1;
  static RenderComponentType _renderType;
  ViewInlinedNgIfWithImmutableConditionComponent0(AppView<dynamic> parentView, int parentIndex) : super(import7.ViewType.COMPONENT, {}, parentView, parentIndex, ChangeDetectionStrategy.CheckAlways, nodeDebugInfos_InlinedNgIfWithImmutableConditionComponent0) {
    rootEl = import4.document.createElement('inlined-ngif-with-immutable-condition');
    _renderType ??= import9.appViewUtils.createRenderType('asset:_goldens/test/_files/inlined_ngif.dart class InlinedNgIfWithImmutableConditionComponent - inline template', ViewEncapsulation.None, styles$InlinedNgIfWithImmutableConditionComponent);
    setupComponentType(_renderType);
  }
  @override
  ComponentRef<import3.InlinedNgIfWithImmutableConditionComponent> build() {
    final _rootEl = rootEl;
    final import4.HtmlElement parentRenderNode = initViewRoot(_rootEl);
    _anchor_0 = ngAnchor.clone(false);
    parentRenderNode.append(_anchor_0);
    dbgElm(this, _anchor_0, 0, 0, 5);
    init([], null, [_anchor_0]);
    return null;
  }

  @override
  void detectChangesInternal() {
    final import3.InlinedNgIfWithImmutableConditionComponent _ctx = ctx;
    bool firstCheck = (this.cdState == 0);
    if (firstCheck) {
      if (!identical(_ctx.foo, null)) {
        if (_ctx.foo) {
          var doc = import4.document;
          _el_0_0 = doc.createElement('div');
          dbgElm(this, _el_0_0, 0, 0, 0);
          _text_0_1 = new import4.Text('Hello World!');
          _el_0_0.append(_text_0_1);
          dbgElm(this, _text_0_1, 1, 0, 17);
          addInlinedNodes(_anchor_0, [_el_0_0], true);
        }
      }
    }
  }
}

AppView<import3.InlinedNgIfWithImmutableConditionComponent> viewFactory_InlinedNgIfWithImmutableConditionComponent0(AppView<dynamic> parentView, int parentIndex) {
  return new ViewInlinedNgIfWithImmutableConditionComponent0(parentView, parentIndex);
}

const List<dynamic> styles$InlinedNgIfWithImmutableConditionComponentHost = const [];
List<StaticNodeDebugInfo> nodeDebugInfos_InlinedNgIfWithImmutableConditionComponentHost0 = [
  new StaticNodeDebugInfo([import3.InlinedNgIfWithImmutableConditionComponent], import3.InlinedNgIfWithImmutableConditionComponent, <String, dynamic>{})
];

class _ViewInlinedNgIfWithImmutableConditionComponentHost0 extends DebugAppView<dynamic> {
  ViewInlinedNgIfWithImmutableConditionComponent0 _compView_0;
  import3.InlinedNgIfWithImmutableConditionComponent _InlinedNgIfWithImmutableConditionComponent_0_5;
  _ViewInlinedNgIfWithImmutableConditionComponentHost0(AppView<dynamic> parentView, int parentIndex) : super(import7.ViewType.HOST, {}, parentView, parentIndex, ChangeDetectionStrategy.CheckAlways, nodeDebugInfos_InlinedNgIfWithImmutableConditionComponentHost0);
  @override
  ComponentRef build() {
    _compView_0 = new ViewInlinedNgIfWithImmutableConditionComponent0(this, 0);
    rootEl = _compView_0.rootEl;
    dbgIdx(rootEl, 0);
    _InlinedNgIfWithImmutableConditionComponent_0_5 = new import3.InlinedNgIfWithImmutableConditionComponent();
    _compView_0.create(_InlinedNgIfWithImmutableConditionComponent_0_5, projectableNodes);
    init0Dbg(rootEl, [rootEl]);
    return new ComponentRef<import3.InlinedNgIfWithImmutableConditionComponent>(0, this, rootEl, _InlinedNgIfWithImmutableConditionComponent_0_5);
  }

  @override
  void detectChangesInternal() {
    _compView_0.detectChanges();
  }

  @override
  void destroyInternal() {
    _compView_0?.destroy();
  }
}

AppView viewFactory_InlinedNgIfWithImmutableConditionComponentHost0(AppView<dynamic> parentView, int parentIndex) {
  return new _ViewInlinedNgIfWithImmutableConditionComponentHost0(parentView, parentIndex);
}

const ComponentFactory<import3.InlinedNgIfWithImmutableConditionComponent> InlinedNgIfWithImmutableConditionComponentNgFactory = const ComponentFactory<import3.InlinedNgIfWithImmutableConditionComponent>('inlined-ngif-with-immutable-condition', viewFactory_InlinedNgIfWithImmutableConditionComponentHost0, _InlinedNgIfWithImmutableConditionComponentMetadata);
const _InlinedNgIfComponentMetadata = const [];
const _InlinedNgIfWithImmutableConditionComponentMetadata = const [];
var _visited = false;
void initReflector() {
  if (_visited) {
    return;
  }
  _visited = true;

  _ngRef.registerComponent(InlinedNgIfComponent, InlinedNgIfComponentNgFactory);
  _ngRef.registerComponent(InlinedNgIfWithImmutableConditionComponent, InlinedNgIfWithImmutableConditionComponentNgFactory);
  _ref0.initReflector();
}
